db/migrate/20240429093317_add_retention_instruction_to_labware.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+class AddRetentionInstructionToLabware < ActiveRecord::Migration[6.1]
+  def up
+    add_column :labware, :retention_instruction, :integer
```

Comment:
Is there a default value for retention_instruction? I think, there should be one and the existing records should be updated in a data migration.

Reply:
We have to think about this. I don't know if there is any type of labware that does not relate to a retention instruction. If there is, I think a default value will break things, since when we're creating the column (i.e., doing the migration) it will automatically set the default value to _every_ labware.

I've chosen not to use a default value because we've got another story to build a script to fix the data in https://github.com/sanger/sequencescape/issues/4095. I have made some changes (such as in [`app/views/labware/_asset_summary.html.erb`](https://github.com/sanger/sequencescape/pull/4096/files#diff-c255eec018f41cb51557c417cd5938f01d2ac4d0a1d1e5a487ae2763ae0a8841R103-R120)) to make this story self-contained. That is, this story was made to be deployed independently without side-effects. This story can be deployed independently without breaking existing stuff.

Reply:
A couple of points:

- After deploying the data migration story, this story will need a revisit to remove the redundant checks on availability of retention instruction in metadata.

- A note about validations:
In the schema migration, the null option is not specified, hence it is nullable.
In sample manifest upload processors, if the retention instruction column is present, the value cannot be blank.
Enforcing a retention instruction value is done by adding a manifest column where necessary.